client/index.js

```diff
@@ -35,7 +36,7 @@ const sox = (args = {}) => {
     query: merge(session, query())
   }
 
-  const socket = io(base, opts)
+  const socket = io(base, merge(options, opts))
```

Comment:
A few comments here:
1.  I'm tempted to say that `opts` in this context is more properly `defaults`, except...
1.  The way `merge` is used here will make these `defaults` override your overrides, which normally I would say is backwards, but...
1.  The `path` and `query` bits are integral to the way that `sox` works at all, so we don't want to override them.

So at that point, I'm not sure what the best direction is.  Thoughts?

Reply:
Thanks for the thoughts, so what I want to do is add options for `randomizationFactor`, `reconnectionDelayMax`, `reconnectionDelay`, to help alleviate some of the stress deployments put on our backend servers.  

So with that context yes I agree that `opts` is more like `defaults` for our use of the `socket.io/client`, `merge` is 100% backwards (🤦‍♂️) and `sox` special sauce is those `path` and `query` above so they should override anything that an implementation may pass in.

Let me massage this a little and see how it feels after.